src/MountedTraversal.js

```diff
@@ -207,8 +214,14 @@ export function buildInstPredicate(selector) {
       }
       break;
 
+    case 'object':
+      if (!isArray(selector) && !isNull(selector)) {
+        return node => instHasProps(node, selector);
+      }
+      throw new TypeError('An array or null is not supported as a selector');
+
     default:
-      throw new TypeError('Expecting a string or Component Constructor');
+      throw new TypeError('Expecting a string, object, or Component Constructor');
```

Comment:
Maybe not for this PR, maybe for this one. It'd be nice if these `TypeError`s was prefaced with something like `Enzyme::SelectorError ...` 

similar to https://github.com/airbnb/enzyme/blob/master/src/ShallowWrapper.js#L107